fl4health/preprocessing/warmed_up_module.py

```diff
@@ -0,0 +1,121 @@
+import json
+import os
+from logging import INFO
+from pathlib import Path
+from typing import Optional
+
+import torch
+from flwr.common.logger import log
+
+
+class WarmedUpModule:
+    """This class is used to load a pretrained model into the current model."""
+
+    def __init__(
+        self,
+        pretrained_model: Optional[torch.nn.Module] = None,
+        pretrained_model_path: Optional[Path] = None,
+        weights_mapping_path: Optional[Path] = None,
+    ) -> None:
+        """Initialize the WarmedUpModule with the pretrained model stats and weights mapping dict.
+
+        Args:
+            pretrained_model (Optional[torch.nn.Module]): Pretrained model.
+                                                          This is mutually exclusive with pretrained_model_path.
+            pretrained_model_path (Optional[Path]): Path of the pretrained model.
+                                                    This is mutually exclusive with pretrained_model.
+            weights_mapping_dir (Optional[str], optional): Path of to json file of the weights mapping dict.
+            If models are not exactly the same, a weights mapping dict is needed to map the weights of the pretrained
+            model to the current model.
+        """
+        if pretrained_model is not None and pretrained_model_path is not None:
+            AssertionError(
+                "pretrained_model_path and pretrained_model is mutually exclusive. Please provide one of them."
+            )
+
+        elif pretrained_model is not None:
+            log(INFO, "Pretrained model is provided.")
+            self.pretrained_model_state = pretrained_model.state_dict()
+
+        elif pretrained_model_path is not None:
+            assert os.path.exists(
+                pretrained_model_path
+            ), f"Pretrained model path {pretrained_model_path} does not exist."
+            log(INFO, f"Loading pretrained model from {pretrained_model_path}")
+            self.pretrained_model_state = torch.load(pretrained_model_path).state_dict()
+
+        else:
+            raise AssertionError("At least one of pretrained_model_path and pretrained_model should be provided.")
+
+        if weights_mapping_path is not None:
+            with open(weights_mapping_path, "r") as file:
+                self.weights_mapping_dict = json.load(file)
+                log(INFO, f"Weights mapping dict: {self.weights_mapping_dict}")
```

Comment:
When the mapping is quite large (think language model), this is going to be a lot to log. Perhaps we just log that it was successfully loaded? If we think it's helpful to log the dictionary, perhaps we can log up to small number of components, like 5-10?

Reply:
I think we can omit it.